app/services/atmosphere/destroy_appliance_set.rb

```diff
@@ -0,0 +1,19 @@
+module Atmosphere
+  class DestroyApplianceSet
+    def initialize(appliance_set)
+      @appliance_set = appliance_set
+    end
+
+    def execute
```

Comment:
Unfortunately not. While deleting appliance destructive operation are performed on compute site (VMs are removed). If transaction would be used here and rollback would be performed we would have missing VMs. 

This is a reason I implemented it using best effort way.